mqtt/mqtt-otel/src/lib.rs

```diff
@@ -0,0 +1,92 @@
+use hyper::{
+    header::CONTENT_TYPE,
+    service::{make_service_fn, service_fn},
+    Body, Error, Request, Response, Server,
+};
+
+use core::future::Future;
+use std::pin::Pin;
+
+use futures::stream::Stream;
+use futures::StreamExt;
+use opentelemetry::metrics::{self, MetricsError};
+use opentelemetry::sdk::metrics::PushController;
+use opentelemetry_prometheus::PrometheusExporter;
+use prometheus::{Encoder, TextEncoder};
+use std::convert::Infallible;
+use std::result::Result;
+use std::time::Duration;
+
+async fn serve_req(
+    _req: Request<Body>,
+    prom_exporter: PrometheusExporter,
+) -> Result<Response<Body>, hyper::Error> {
+    let mut buffer = vec![];
+    let encoder = TextEncoder::new();
+    let metric_families = prom_exporter.registry().gather();
+    encoder.encode(&metric_families, &mut buffer).unwrap();
+
+    let response = Response::builder()
+        .status(200)
+        .header(CONTENT_TYPE, encoder.format_type())
+        .body(Body::from(buffer))
+        .unwrap();
+
+    Ok(response)
+}
+
+pub fn init_prometheus_metrics_exporter() -> Result<PrometheusExporter, MetricsError> {
+    opentelemetry_prometheus::exporter().try_init()
+}
+
+pub fn create_prometheus_server(
```

Comment:
This will not integrate very well with the EdgeHub container. We need to discuss how to do that..

Reply:
Will include mention/depiction of this in the design docs/diagrams.  These docs/diagrams should help with the discussion about the path forward integrating with the EdgeHub container.